ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/SentenceTransformerTranslator.java

```diff
@@ -25,7 +25,12 @@ public Batchifier getBatchifier() {
     @Override
     public void prepare(TranslatorContext ctx) throws IOException {
         Path path = ctx.getModel().getModelPath();
-        tokenizer = HuggingFaceTokenizer.builder().optPadding(true).optTokenizerPath(path.resolve("tokenizer.json")).build();
+        tokenizer = HuggingFaceTokenizer
+            .builder()
+            .optPadding(true)
+            .optTruncation(true)
```

Comment:
Why add this?

Reply:
@HenryL27 Isn't it same when we add truncation information in the tokenizer.json file like this:

```
"truncation": {
    "direction": "Right",
    "max_length": 128,
    "strategy": "LongestFirst",
    "stride": 0
  },
  ```
  
 Can you please share your tokenizer file you are using?

Reply:
If I don't and the context is too long, I get sometimes index-out-of-bounds when using a roberta-based model to infer (there are 514 learned positional embeddings so if you go over the djl engine breaks)

Reply:
you're right, I've got 
```
"truncation": null
```

Reply:
Yup, which is why we added [this method ](https://github.com/opensearch-project/opensearch-py-ml/blob/main/opensearch_py_ml/ml_models/sentencetransformermodel.py#L735-L763) to update the tokenizer.json file for our pre-trained model. 

I feel like in future, we should also remove this padding from here, as that info should also come from tokenizer.json file.

Reply:
why shouldn't we do this though?

Reply:
what if customer doesn't want the truncation? As a platform, how can we support this?